components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -13,292 +22,241 @@ import ChannelInviteModal from 'components/channel_invite_modal';
 import SidebarMenu from 'components/sidebar/sidebar_menu';
 import Menu from 'components/widgets/menu/menu';
 
-import {ModalData} from 'types/actions';
-
 import Constants, {ModalIdentifiers} from 'utils/constants';
 import {copyToClipboard} from 'utils/utils';
 
-type Props = {
-    channel: Channel;
-    channelLink: string;
-    currentUserId: string;
-    currentTeamId: string;
-    isUnread: boolean;
-    isFavorite: boolean;
-    isMuted: boolean;
-    intl: IntlShape;
-    managePublicChannelMembers: boolean;
-    managePrivateChannelMembers: boolean;
-    closeHandler?: (callback: () => void) => void;
-    isCollapsed: boolean;
-    isMenuOpen: boolean;
-    onToggleMenu: (isMenuOpen: boolean) => void;
-    actions: {
-        markMostRecentPostInChannelAsUnread: (channelId: string) => void;
-        markChannelAsRead: (channelId: string) => void;
-        favoriteChannel: (channelId: string) => void;
-        unfavoriteChannel: (channelId: string) => void;
-        muteChannel: (userId: string, channelId: string) => void;
-        unmuteChannel: (userId: string, channelId: string) => void;
-        openModal: <P>(modalData: ModalData<P>) => void;
-    };
-};
-
-type State = {
-    openUp: boolean;
-};
-
-export class SidebarChannelMenu extends React.PureComponent<Props, State> {
-    isLeaving: boolean;
+import type {PropsFromRedux, OwnProps} from './index';
 
-    constructor(props: Props) {
-        super(props);
+type Props = PropsFromRedux & OwnProps;
 
-        this.state = {
-            openUp: false,
-        };
+const SidebarChannelMenu = (props: Props) => {
+    const isLeaving = useRef(false);
+    const [openUp, setOpenUp] = useState(false);
 
-        this.isLeaving = false;
-    }
+    const {formatMessage} = useIntl();
 
-    markAsRead = () => {
-        this.props.actions.markChannelAsRead(this.props.channel.id);
+    function handleMarkAsRead() {
```

Comment:
Most of these functions could use a `useCallback`, since they are used in props. If the receiving component is well known and "small" (not much more than a div) you can leave it as it is, but I think it would not hurt to have the use callbacks, just in case.

Reply:
the new menus is pretty simple div tree structure. Will take a look at for using the callback in the new one